console/console-init/ui/src/pages/CreateAddress/CreateAddressDefinition.tsx

```diff
@@ -164,28 +178,26 @@ export const AddressDefinition: React.FunctionComponent<IAddressDefinition> = ({
       setTopic(event.currentTarget.childNodes[0].value);
     setIsTopicOpen(!isTopicOpen);
   };
-  const { loading, error, data } = useQuery<IAddressTypes>(
-    RETURN_ADDRESS_TYPES,
-    {
-      variables: {
-        a: addressSpaceType
-      }
-    }
-  );
-  if (loading) return <Loading />;
-  if (error) return <Loading />;
-  const { addressTypes_v2 } = data || {
-    addressTypes_v2: []
-  };
-  const types: IDropdownOption[] = addressTypes_v2.map(type => {
-    return {
-      value: type.spec.displayName,
-      label: type.spec.displayName,
-      description: type.spec.shortDescription
-    };
-  });
-  if (typeOptions.length === 0) setTypeOptions(types);
-
+  if (addressSpaceType) {
+    const addressTypes = client.query<IAddressTypes>({
```

Comment:
is it possible to wrap this  api code in function like getAddressType and call this function.